qiita_db/metadata_template.py

```diff
@@ -1561,9 +1561,16 @@ def update(self, md_template):
                                % ', '.join(samples_diff))
         columns_diff = set(new_map.columns) - set(current_map.columns)
         if columns_diff:
-            raise QiitaDBError('The new sample template differs from what is '
-                               'stored in database by these columns names: %s'
-                               % ', '.join(columns_diff))
+            queue = 'add_cols_%d' % self._id
+            datatypes = dict(izip(new_map.columns, _get_datatypes(new_map)))
+            conn_handler.create_queue(queue)
+
+            for col in columns_diff:
+                self.add_category(col, new_map[col].to_dict(), datatypes[col])
```

Comment:
This is how the old code did it (look at line 1587). If we want a full re-factor that is out of the scope of this pull request, we just need the functionality quickly for EBI submissions. Issue .#1020 created for a full refactor.